examples/step-87/step-87.cc

```diff
@@ -0,0 +1,1175 @@
+    DistributedTriangulation<dim> tria(MPI_COMM_WORLD);
+    GridGenerator::subdivided_hyper_cube(tria, 7);
+
+    FE_Q<dim>       fe(fe_degree);
+    DoFHandler<dim> dof_handler(tria);
+    dof_handler.distribute_dofs(fe);
+
+    // We determine a finite element solution representing implicitly
+    // the geometry of a sphere with a radius of $r=0.25$ and the center at
+    // $(0.5,0.5)$ via a signed distance function.
+    LinearAlgebra::distributed::Vector<double> signed_distance;
+    signed_distance.reinit(dof_handler.locally_owned_dofs(),
+                           DoFTools::extract_locally_active_dofs(dof_handler),
+                           MPI_COMM_WORLD);
+
+    VectorTools::interpolate(
+      mapping,
+      dof_handler,
+      Functions::SignedDistance::Sphere<dim>(Point<dim>(0.5, 0.5), 0.25),
+      signed_distance);
+
+    // Next, we fill a vector from an arbitrary function that should represent
+    // a possible finite element solution, which we would like to evaluate.
+    LinearAlgebra::distributed::Vector<double> solution;
+    solution.reinit(dof_handler.locally_owned_dofs(),
+                    DoFTools::extract_locally_active_dofs(dof_handler),
+                    MPI_COMM_WORLD);
+
+    VectorTools::interpolate(mapping,
+                             dof_handler,
+                             Functions::SignedDistance::Plane<dim>(
+                               Point<dim>(), Point<dim>::unit_vector(0)),
+                             solution);
+
+    // We create a list of arbitrary (evaluation) points along a horizontal
+    // line, which intersects the center of the sphere. We do this only
+    // on the root rank, since we intend to output the results to a CSV file
+    // by the root rank.
+    std::vector<Point<dim>> profile;
+    if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0)
+      profile = create_points_along_line(Point<dim>(0.0, 0.5),
+                                         Point<dim>(1.0, 0.5),
+                                         20);
+
+    // Now, we can evaluate the results, e.g., for the signed distance
+    // function at all evaluation points in one go. First, we create a
+    // modifiable Utilities::MPI::RemotePointEvaluation object. We use
+    // VectorTools::point_values() by specifying the number of components of the
+    // solution vector (1 for the present example) as a template parameter.
+    // Within this function, the provided object for
+    // Utilities::MPI::RemotePointEvaluation is automatically reinitialized with
+    // the given points (profile). The ghost values of the to be read vector
+    // need to be updated from the user.
+    Utilities::MPI::RemotePointEvaluation<dim, dim> rpe;
+
+    signed_distance.update_ghost_values();
+    const std::vector<double> profile_signed_distance =
+      VectorTools::point_values<1>(
+        mapping, dof_handler, signed_distance, profile, rpe);
+    signed_distance.zero_out_ghost_values();
```

Comment:
Can you add a comment why this needs to be zeroed out?

Reply:
... related to your previous comment. It would not be necessary to call `zero_out_ghost_values()` here.